include/dali/core/mm/pool_resource.h

```diff
@@ -100,6 +322,18 @@ class pool_resource_base : public memory_resource<kind, Context> {
     }
   }
 
+  /**
+   * @brief Deallocates a block memory without synchronization
+   *
+   * Places a block of memory in the free list for immediate reuse.
+   * The caller must guarantee, that the memory is available without
+   * any additional synchronization in the execution context for this resource.
+   */
+  void deallocate_no_sync(void *ptr, size_t bytes, size_t alignment = alignof(std::max_align_t)) {
```

Comment:
I have to admit that I don't like this function very much. I just couldn't find a better alternative if I want to have both synchronized and unsynchronized deallocation in one resource.